File: Kubex.Models/Post.cs
using System.Collections.Generic;

namespace Kubex.Models
{
    public class Post
    {
        public int Id { get; set; }
        public string Name { get; set; }

        public int? CompanyId { get; set; }
        public virtual Company Company { get; set; }

        public int? AddressId { get; set; }
        public virtual Address Address { get; set; }

        public int? LocationId { get; set; }
        public virtual Location Location { get; set; }

        public virtual ICollection<Contact> Contacts { get; set; }
        public virtual ICollection<Team> Teams { get; set; }
    }
}

File: Kubex.SPA/src/app/_models/post.ts
import { DailyActivityReport } from './dailyActivityReport';
import { Address } from './address';
import { Company } from './company';
import { User } from '.';

export class Post {
    id?: number;
    name: string;
    companyId: number;
    company?: Company;
    address?: Address;
    location?: Location;
    dailyActivityReports?: DailyActivityReport[];
    users?: User[];
}
